src/psbt/mod.rs

```diff
@@ -37,15 +48,39 @@ impl PsbtUtils for Psbt {
             None
         }
     }
+
+    fn fee_amount(&self) -> Option<u64> {
+        let tx = &self.unsigned_tx;
+        let utxos: Option<Vec<TxOut>> = (0..tx.input.len()).map(|i| self.get_utxo_for(i)).collect();
+
+        utxos.map(|inputs| {
+            let input_amount = inputs.iter().fold(0, |acc, i| acc + i.value);
+
+            let output_amount = &self
+                .unsigned_tx
+                .output
+                .iter()
+                .fold(0, |acc, o| acc + o.value);
+
+            input_amount - output_amount
```

Comment:
Just to be extra careful I would either add an assertion that `input_amount` is greater than or equal `output_amount`, or you could use `checked_sub().expect()` to achieve the same result.

It should never happen, but PSBTs are untrusted input so I think it's better to be careful.

Reply:
Safety first, fixed.